### PR TITLE
moved the Toggle_Middle/Shift Clicking from IC verb to GamePref

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Allows to save your preference of using shift key instead of middle click button to activate abilities as a xenos. 

## Why It's Good For The Game

Too many times i forgot to switch back to shift key, having to do that each time you spawn a new xenos is a pain in the ass. A better solution would be to put that settings in keybinding, but i'm not experienced enough to do that

## Changelog
:cl:
tweak: moved the Toggle_Middle/Shift Clicking from IC to GamePref
/:cl:

### CODE REVIEW

**Comment by BraveMole** [comment]
Tested on my own server, i think i did everything right regarding saved files, but a mistake on these is possible. I didn't try loading a savefile from the main branch to try if it works, cause i don't know how

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -28,6 +28,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	var/toggles_chat = TOGGLES_CHAT_DEFAULT
 	var/toggles_sound = TOGGLES_SOUND_DEFAULT
 	var/toggles_gameplay = TOGGLES_GAMEPLAY_DEFAULT
+	var/middleshift_button=TRUE
```
Comment (Hughgent) [inline code/modules/client/preferences.dm:31]: This should not be its own var. it should be part of `toggles_gameplay`

**Review by Hughgent — CHANGES_REQUESTED** [review]

File: code/modules/client/preferences.dm (hunk 1 above)
Comment (BraveMole) [reply code/modules/client/preferences.dm]: Done

File: code/modules/client/preferences_savefile.dm [hunk 2]
```diff
@@ -254,6 +254,7 @@
 	WRITE_FILE(S["auto_fit_viewport"], auto_fit_viewport)
 	WRITE_FILE(S["menuoptions"], menuoptions)
 	WRITE_FILE(S["key_bindings"], key_bindings)
+
```
Comment (Hughgent) [inline code/modules/client/preferences_savefile.dm:257]: not needed

File: code/modules/client/preferences_savefile.dm [hunk 3]
```diff
@@ -266,6 +267,7 @@
 	WRITE_FILE(S["mute_self_combat_messages"], mute_self_combat_messages)
 	WRITE_FILE(S["mute_others_combat_messages"], mute_others_combat_messages)
 
+
```
Comment (Hughgent) [inline code/modules/client/preferences_savefile.dm:270]: Not needed

**Review by Hughgent — CHANGES_REQUESTED** [review]
getting better

File: code/__DEFINES/preferences.dm [hunk 4]
```diff
@@ -65,8 +65,9 @@
 #define TOGGLES_DEADCHAT_DEFAULT (NONE)
 
 #define RADIAL_MEDICAL (1<<0)
+#define MIDDLESHIFTCLICKING (1<<1)
```
Comment (Hughgent) [inline code/__DEFINES/preferences.dm:68]: I would appreciate if these were both tabbed out to some common level. Just so long as they are both on the same `(1<<0)` level, not unlike other bitshift defines.

**Review by Hughgent — APPROVED** [review]

File: code/__DEFINES/preferences.dm (hunk 4 above)
Comment (BraveMole) [reply code/__DEFINES/preferences.dm:68]: tabbed out? I don't understand what you are asking

Comment (TiviPlus) [reply code/__DEFINES/preferences.dm:68]: Dont

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**